Add DynamoDB config parameters override samples (#5305)
aws-amplify/docs: src/pages/cli/graphql/override.mdx

  resources.models["Todo"].modelDatasource.dynamoDbConfig['deltaSyncConfig']['baseTableTtl'] = '3600'
}
```
### Example - Configure DynamoDBBillingMode

Set the DynamoDB billing mode for the API. One of "PROVISIONED" or "PAY_PER_REQUEST".

```ts
import { AmplifyApiGraphQlResourceStackTemplate } from '@aws-amplify/cli-extensibility-helper';

export function override(resources: AmplifyApiGraphQlResourceStackTemplate) {
  resources.models['Post'].modelDDBTable.billingMode = 'PAY_PER_REQUEST'
}
```
### Example - Configure DynamoDBModelTableProvisionedThroughput
Override the default [ProvisionedThroughput](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-provisionedthroughput) provisioned for each `@model` table

> Only valid if the "DynamoDBBillingMode" is set to "PROVISIONED"

Reviewer: Once we change the table's billing mode to "PROVISIONED", I believe we need to provide provisionedThroughput for all the GSIs on the table too. It would be nice to verify and add that info.

```ts
import { AmplifyApiGraphQlResourceStackTemplate } from '@aws-amplify/cli-extensibility-helper';

export function override(resources: AmplifyApiGraphQlResourceStackTemplate) {
  resources.models['Post'].modelDDBTable.provisionedThroughput = {
    readCapacityUnits:5,
    writeCapacityUnits:5
  }
}
```

### Example - Configure DynamoDBEnablePointInTimeRecovery
Enable/disable [DynamoDB Point-in-time-Recovery](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/PointInTimeRecovery.html) for all model tables

```ts
import { AmplifyApiGraphQlResourceStackTemplate } from '@aws-amplify/cli-extensibility-helper';

export function override(resources: AmplifyApiGraphQlResourceStackTemplate) {
  resources.models['Post'].modelDDBTable.pointInTimeRecoverySpecification = {

>     pointInTimeRecoveryEnabled:true

Reviewer 2: ```suggestion
    pointInTimeRecoveryEnabled: true
```

  resources.models["Todo"].modelDatasource.dynamoDbConfig['deltaSyncConfig']['baseTableTtl'] = '3600'
}
```
### Example - Configure DynamoDBBillingMode

> Set the DynamoDB billing mode for the API. One of "PROVISIONED" or "PAY_PER_REQUEST".

Reviewer 2: ```suggestion
Set the [DynamoDB billing mode](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BillingModeSummary.html) for the API. One of "PROVISIONED" or "PAY_PER_REQUEST".
```

since we link to Point-in-time-recovery I think it would be helpful to link each of these properties we are overriding

```ts
import { AmplifyApiGraphQlResourceStackTemplate } from '@aws-amplify/cli-extensibility-helper';

export function override(resources: AmplifyApiGraphQlResourceStackTemplate) {
  resources.models['Post'].modelDDBTable.billingMode = 'PAY_PER_REQUEST'
}
```
### Example - Configure DynamoDBModelTableProvisionedThroughput

> Override the default IOPS provisioned for each @model table

Reviewer 2: ```suggestion
Override the default IOPS provisioned for each `@model` table
```

encapsulate directive reference with an inline code block

Reviewer 2: what does IOPS stand for? Can we include its name followed by the acronym, including a link to relevant docs?

Only valid if the "DynamoDBBillingMode" is set to "PROVISIONED"

```ts
import { AmplifyApiGraphQlResourceStackTemplate } from '@aws-amplify/cli-extensibility-helper';

export function override(resources: AmplifyApiGraphQlResourceStackTemplate) {
  resources.models['Post'].modelDDBTable.provisionedThroughput ={
    readCapacityUnits:5,
    writeCapacityUnits:5
  }
}
```

### Example - Configure DynamoDBEnablePointInTimeRecovery

> Enable/disable DynamoDB Point-in-time-Recovery for all model tables
> 
> For more information, see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/PointInTimeRecovery.html

Reviewer 2: ```suggestion
Enable/disable [DynamoDB Point-in-time-Recovery](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/PointInTimeRecovery.html) for all model tables
```

Reviewer 2: to ease reading and prevent raw links

  resources.models["Todo"].modelDatasource.dynamoDbConfig['deltaSyncConfig']['baseTableTtl'] = '3600'
}
```

> ### Example - Configure DynamoDBBillingMode

Reviewer 3: ```suggestion
### Example - Configure DynamoDB table's billing mode
```

Set the [DynamoDB billing mode](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-billingmode) for the API. One of "PROVISIONED" or "PAY_PER_REQUEST".

```ts
import { AmplifyApiGraphQlResourceStackTemplate } from '@aws-amplify/cli-extensibility-helper';

export function override(resources: AmplifyApiGraphQlResourceStackTemplate) {

>   resources.models['Post'].modelDDBTable.billingMode = 'PAY_PER_REQUEST'

Reviewer 4: ```suggestion
  resources.models['Post'].modelDDBTable.billingMode = 'PAY_PER_REQUEST';
```

I think a lot of code samples in this file can use the `;`

}
```

> ### Example - Configure DynamoDBModelTableProvisionedThroughput

Reviewer 3: ```suggestion
### Example - Configure provisioned throughput for DynamoDB table or Global Secondary Index
```

> Override the default [ProvisionedThroughput](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-provisionedthroughput) provisioned for each `@model` table

Reviewer 4: ```suggestion
Override the default [ProvisionedThroughput](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-dynamodb-table.html#cfn-dynamodb-table-provisionedthroughput) provisioned for each `@model` table and its Global Secondary Indexes (GSI).
```

Only valid if the "DynamoDBBillingMode" is set to "PROVISIONED"

```ts
import { AmplifyApiGraphQlResourceStackTemplate } from '@aws-amplify/cli-extensibility-helper';

export function override(resources: AmplifyApiGraphQlResourceStackTemplate) {
  resources.models['Post'].modelDDBTable.provisionedThroughput = {

>     readCapacityUnits : 5,
>     writeCapacityUnits : 5

Reviewer 2: ```suggestion
    readCapacityUnits: 5,
    writeCapacityUnits: 5
```

  }
  
  /**

>   * It is necessary to include provisionedThroughput for every Global Secondary Index (GSI) that exists in the table.

Reviewer: ```suggestion
  * It is necessary to set provisionedThroughput for every Global Secondary Index (GSI) that exists in the table.
```

>   * When billing mode is set to "PROVISIONED", It is necessary to specify provisionedThroughput for every Global Secondary Index (GSI) that exists in the table.

Reviewer 4: ```suggestion
  * When billing mode is set to "PROVISIONED", it is necessary to specify `provisionedThroughput` for every Global Secondary Index (GSI) that exists in the table.
```

  */
  
  resources.models[
    "Report"
  ].modelDDBTable.globalSecondaryIndexes[0].provisionedThroughput = {
    readCapacityUnits: 5,
    writeCapacityUnits: 5

>   }; 

Reviewer 4: Any reason this code sample used `Report' model instead of 'Post`?

>   resources.models[
>     "Post"
>   ].modelDDBTable.globalSecondaryIndexes[0].provisionedThroughput = {
>     readCapacityUnits: 5,
>     writeCapacityUnits: 5
>   }; 
> }

Reviewer 3: ```suggestion
  resources.models["Post"].modelDDBTable.globalSecondaryIndexes[0].provisionedThroughput = {
    readCapacityUnits: 5,
    writeCapacityUnits: 5
  }; 
}
```

```

> ### Example - Configure DynamoDBEnablePointInTimeRecovery

Reviewer 3: ```suggestion
### Example - Enable point-in-time recovery for DynamoDB table
```

> Enable/disable [DynamoDB Point-in-time-Recovery](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-pointintimerecoveryspecification.html) for all model tables

Reviewer 4: ```suggestion
Enable/disable [DynamoDB Point-in-time-Recovery](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-pointintimerecoveryspecification.html) for each `@model` table.
```

> Enable/disable [DynamoDB Point-in-time-Recovery](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-pointintimerecoveryspecification.html) for each `@model` table.

Reviewer 3: ```suggestion
Enable/disable [DynamoDB point-in-time recovery](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-dynamodb-table-pointintimerecoveryspecification.html) for each `@model` table.
```

```ts
import { AmplifyApiGraphQlResourceStackTemplate } from '@aws-amplify/cli-extensibility-helper';

export function override(resources: AmplifyApiGraphQlResourceStackTemplate) {
  resources.models['Post'].modelDDBTable.pointInTimeRecoverySpecification = {

>     pointInTimeRecoveryEnabled : true

Reviewer 2: ```suggestion
    pointInTimeRecoveryEnabled: true
```
